File: leetcode/cpp/partition-array-according-to-given-pivot.cpp
#include <bits/stdc++.h>
using namespace std;
class Solution
{
public:
    vector<int> pivotArray(vector<int> &nums, int pivot)
    {
        // place all elements less than pivot to left
        int i = 0, j = 0, n = nums.size();
        while (j < n)
        {
            if (nums[j] < pivot)
            {
                swap(nums[i], nums[j]);
                i += 1;
            }
            j += 1;
        }
        // place all the elements greater than pivot to right
        i = n - 1, j = n - 1;
        while (j >= 0)
        {
            if (nums[j] > pivot)
            {
                swap(nums[i], nums[j]);
                i -= 1;
            }
            j -= 1;
        }
        return nums;
    }
};

File: leetcode/java/PartitionArrayAccordingtoGivenPivot.java
import java.util.ArrayList;
import java.util.List;

public class PartitionArrayAccordingtoGivenPivot {

    public int[] pivotArray(int[] nums, int pivot){
        int n = nums.length;
        int[] answer = new int[n];
        for(int j = 0 ; j < n; j++) answer[j] = pivot;
        int j = 0;
        for(int i = 0; i < n; i++){
            if(nums[i] < pivot){
                answer[j++] = nums[i];
            }
        }
        j = n-1;
        for(int i = n-1; i >= 0; i--){
            if(nums[i]>pivot){
                answer[j--] = nums[i];
            }
        }
        return answer;
    }
}